Utilities/JITASM.cpp

```diff
@@ -185,8 +185,13 @@ static u8* add_jit_memory(usz size, usz align)
 	if (olda != newa) [[unlikely]]
 	{
 #ifndef CAN_OVERCOMMIT
-		// Commit more memory
-		utils::memory_commit(pointer + olda, newa - olda, Prot);
+		// Commit more memory.
+		// NOTE: Calling memory commit in parallel on the same addresses can throw a permission error.
+		{
+			static std::mutex mcommit_lock;
+			std::lock_guard lock(mcommit_lock);
+			utils::memory_commit(pointer + olda, newa - olda, Prot);
```

Comment:
hmm, this should somehow be implemented insidr memory_commit

Reply:
Beware of overlapping regions as well, though I doubt that's a problem in practice.
This operation is pretty quick so I haven't experienced any slowdown. It can also be copied as-is into the memory_commit function.